SmartDeviceLink/SDLMutableDataQueue.h

```diff
@@ -0,0 +1,38 @@
+//
+//  SDLMutableDataQueue.h
+//  
+//
+//  Created by David Switzer on 3/2/17.
+//
+//
+
+#import <Foundation/Foundation.h>
+
+@interface SDLMutableDataQueue : NSObject
+
+NS_ASSUME_NONNULL_BEGIN
+
+// frontBuffer returns the NSMutableData * buffer at the front of the queue,
+// but does not remove it -- modeled after the STL C++ queue front method
+- (NSMutableData *)frontBuffer;
+
+// popBuffer removes the buffer at the head of the queue -- modeled after the
+// STL C++ queue pop method
+- (void)popBuffer;
+
+// Enqueues a new NSMutableData * buffer at the back of the queue
+- (void)enqueueBuffer:(NSMutableData *)data;
+
+// Empty the queue
+- (void)flush;
```

Comment:
When I think of flush, I usually think of it returning a bunch of data. Would this be more accurate as `clear`?